### PR TITLE
Release to SDKMan when publishing a release

### PR DESCRIPTION
**Pull request by eygraber**
## Description

Follows the process laid out in the SDKMan [vendor onboarding process](https://github.com/sdkman/sdkman-cli/wiki/Vendor-onboarding-process) and [gradle plugin](https://github.com/sdkman/sdkman-vendor-gradle-plugin) to push releases of ktlint to SDKMan.

Fixes https://github.com/pinterest/ktlint/issues/1363

## Checklist

- [x] PR description added
- [ ] tests are added
- [ ] KtLint has been applied on source code itself and violations are fixed
- [ ] [documentation](https://pinterest.github.io/ktlint/) is updated
- [ ] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] Rule is added to [rules documentation](https://pinterest.github.io/ktlint/rules/standard/)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -35,6 +35,19 @@ jobs:
...
+        if: ${{ success() }}
+        run: echo "version=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: Create zip for sdkman
+        if: ${{ success() }}
+        run: |
+          cd ktlint-cli/build/run
+          mkdir -p ktlint-${{ env.version }}/bin
+          cp ktlint ktlint-${{ env.version }}/bin
+          zip -rm ktlint-${{ env.version }}.zip ktlint-${{ env.version }}
```
Comment (eygraber) [inline .github/workflows/release.yml:49]: Doing this for https://github.com/sdkman/sdkman-cli/wiki/Well-formed-SDK-archives

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -48,19 +61,22 @@ jobs:
...
         with:
           formula-name: ktlint
-          download-url: https://github.com/pinterest/ktlint/releases/download/${{ env.version }}/ktlint
+          download-url: https://github.com/pinterest/ktlint/releases/download/${{ env.version }}/ktlint-${{ env.version }}.zip
+
+      - name: Release to sdkman
+        if: ${{ success() }}
+        env:
+          SDKMAN_KEY: ${{ secrets.SDKMAN_KEY }}
+          SDKMAN_TOKEN: ${{ secrets.SDKMAN_TOKEN }}
```
Comment (eygraber) [inline .github/workflows/release.yml:77]: @paul-dingemans these need to be added to the repro action secrets after receiving them from https://github.com/sdkman/sdkman-cli/wiki/Vendor-onboarding-process#request-api-credentials

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -35,6 +35,19 @@ jobs:
         if: ${{ success() }}
         uses : ffurrer2/extract-release-notes@v1
 
+      - name: Get version
+        id: get_version
+        if: ${{ success() }}
+        run: echo "version=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: Create zip for sdkman
```
Comment (paul-dingemans) [inline .github/workflows/release.yml:43]: The zip-file is also used for uploading to brew. Please rename task to reflect this, smth like `Create zip for dependency managers`

File: .github/workflows/release.yml (hunk 1 above)
Comment (paul-dingemans) [reply .github/workflows/release.yml:49]: Please add this as a comment before line 47

File: .github/workflows/release.yml (hunk 2 above)
Comment (shashachu) [reply .github/workflows/release.yml:78]: got it. just sent the credentials request.

**Comment by eygraber** [comment]
Awesome, thanks!

@paul-dingemans maybe instead of trying to get the 0.49.0 release integrated, we just try it out with 0.49.1?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -40,12 +40,13 @@ jobs:
...
 
-      - name: Create zip for sdkman
+      - name: Create zip for dependency managers
         if: ${{ success() }}
         run: |
           cd ktlint-cli/build/run
+          # Doing this for Homebrew and https://github.com/sdkman/sdkman-cli/wiki/Well-formed-SDK-archives
           mkdir -p ktlint-${{ env.version }}/bin
-          cp ktlint ktlint-${{ env.version }}/bin
+          cp ktlint ktlint-${{ env.version }}/bin/ktlint-${{ env.version }}
```
Comment (paul-dingemans) [inline .github/workflows/release.yml:49]: I just see that the name of the binary is also suffixed with the version number. That does not seem to be a requirement according to https://github.com/sdkman/sdkman-cli/wiki/Well-formed-SDK-archives. Also, I wonder what the effect could be for the `homebrew` dependency. Please explain why the version number is required in the binary.

**Comment by paul-dingemans** [comment]
Also the changelog and the ktlint documentation needs to be updated.

File: .github/workflows/release.yml (hunk 4 above)
Comment (eygraber) [reply .github/workflows/release.yml:49]: You're right. I was looking at a few other projects and they were doing that, but it isn't a requirement. I'll roll that back.

**Comment by paul-dingemans** [comment]
Tnx for your contribution.

**Pull request merged**